cognite_toolkit/_cdf_tk/prototypes/_packages/update_references.py

```diff
@@ -0,0 +1,338 @@
+import inspect
+import itertools
+import re
+import shutil
+
+# from cognite.client.data_classes.transformations.common import TransformationDestination
+from pathlib import Path
+from typing import Any
+
+import cognite.client.data_classes as data_classes
+import cognite.client.data_classes.data_modeling as dm_data_classes
+import yaml
+from cognite.client.data_classes.data_modeling import ViewApply, ViewId
+from cognite.client.data_classes.iam import Capability
+from cognite.client.data_classes.transformations import OidcCredentials
+from cognite.client.data_classes.transformations.common import TransformationDestination
+from cognite.client.utils._text import to_camel_case
+
+from cognite_toolkit._cdf_tk.loaders import ResourceLoader
+from cognite_toolkit._cdf_tk.validation import validate_resource_yaml
+
+
+# * in a line signifies that the line is a comment, for example where the config has multiple options
+def comment_optional(yaml_str: str) -> str:
+    n = []
+    for line in yaml_str.split("\n"):
+        if "*" in line:
+            n.append(f"#{line}".replace("*", "").replace("'", "").replace(" - :", ""))
+        else:
+            n.append(line.replace("*", ""))
+    return "\n".join(n)
+
+
+IGNORED_ANNOTATIONS = [
+    "str",
+    # "int",
+    "float",
+    "bool",
+    "dict",
+    "list",
+    "None",
+    "enum",
+    "dict[str, str]",
+    "list[str]",
+    "Any",
+    "Filter",
+    "list[ExternalId]",
+    "'allUserAccounts'",
+    "list[ViewId]",
+    "dict[str, ViewPropertyApply]",
+    # "list[ViewId | ViewApply]",
+]
```

Comment:
O(1) better than O(n) Lookup
```suggestion
IGNORED_ANNOTATIONS = {
    "str",
    # "int",
    "float",
    "bool",
    "dict",
    "list",
    "None",
    "enum",
    "dict[str, str]",
    "list[str]",
    "Any",
    "Filter",
    "list[ExternalId]",
    "'allUserAccounts'",
    "list[ViewId]",
    "dict[str, ViewPropertyApply]",
}
```